server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/CassandraBlobExistenceTesterIntegrationTest.java

```diff
@@ -0,0 +1,134 @@
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+@ExtendWith(DockerAwsS3Extension.class)
+class CassandraBlobExistenceTesterIntegrationTest implements MetricableBlobStoreContract {
+    private static final HashBlobId.Factory BLOB_ID_FACTORY = new HashBlobId.Factory();
+
+    @RegisterExtension
+    static CassandraClusterExtension cassandraCluster = new CassandraClusterExtension(CassandraBlobExistenceModule.MODULE);
+
+    ObjectStorageBlobStore objectStorageBlobStore;
+    AwsS3ObjectStorage awsS3ObjectStorage;
+    CassandraBlobExistenceTester cassandraBlobExistenceTester;
+    BlobStore testee;
+
+    @BeforeEach
+    void setUp(DockerAwsS3Container dockerAwsS3, CassandraCluster cassandra) {
+        CassandraBlobExistenceTesterDAO blobExistenceTesterDAO = new CassandraBlobExistenceTesterDAO(BLOB_ID_FACTORY, cassandra.getConf());
+        cassandraBlobExistenceTester = new CassandraBlobExistenceTester(blobExistenceTesterDAO);
+        awsS3ObjectStorage = new AwsS3ObjectStorage(new FakeBlobExistenceTester());
+        AwsS3AuthConfiguration configuration = AwsS3AuthConfiguration.builder()
+            .endpoint(dockerAwsS3.getEndpoint())
+            .accessKeyId(DockerAwsS3Container.ACCESS_KEY_ID)
+            .secretKey(DockerAwsS3Container.SECRET_ACCESS_KEY)
+            .build();
+
+        ObjectStorageBlobStoreBuilder.ReadyToBuild builder = ObjectStorageBlobStore
+            .builder(configuration)
+            .blobIdFactory(BLOB_ID_FACTORY)
+            .blobExistenceTester(cassandraBlobExistenceTester)
+            .blobPutter(awsS3ObjectStorage.putBlob(configuration));
+
+        objectStorageBlobStore = builder.build();
+        testee = new MetricableBlobStore(metricsTestExtension.getMetricFactory(), objectStorageBlobStore);
+    }
+
+    @AfterEach
+    void tearDown() {
+        objectStorageBlobStore.deleteAllBuckets().block();
+        objectStorageBlobStore.close();
+        awsS3ObjectStorage.tearDown();
+    }
+
+    @Override
+    public BlobStore testee() {
+        return testee;
+    }
+
+    @Override
+    public BlobId.Factory blobIdFactory() {
+        return BLOB_ID_FACTORY;
+    }
+
+    @Override
+    @Disabled("JAMES-2829 Unstable with scality/S3 impl")
+    public void readShouldNotReadPartiallyWhenDeletingConcurrentlyBigBlob() {
+
+    }
+
+    @Override
+    @Disabled("JAMES-2838 Unstable with scality/S3 impl")
+    public void readBytesShouldNotReadPartiallyWhenDeletingConcurrentlyBigBlob() {
+
+    }
+
+    @RepeatedTest(100)
+    void deleteShouldNotAffectSubsequentSaveConsistency() throws Exception {
+        BucketName defaultBucketName = testee.getDefaultBucketName();
+        BlobId blobId = blobIdFactory().forPayload(SHORT_BYTEARRAY);
+
+        ConcurrentTestRunner.builder()
+            .reactorOperation(((threadNumber, step) -> {
+                if (step % 2 == 0) {
+                    return testee().save(defaultBucketName, SHORT_BYTEARRAY);
```

Comment:
I don't get it :thinking: , you didn't use `.block()` but at the end, the blob is still in the BlobStore

Reply:
It's declared in a `reactorOperation`, that does a `.block()` at the end if you look at the code